pkg/channelmetrics/observablechan.go

```diff
@@ -76,7 +88,7 @@ func (oc *ObservableChan[T]) Recv(ctx context.Context) T {
 		ctx.Logger().Error(err, "failed to read item from observable channel")
 	}
 
-	return item
+	return item, nil
```

Comment:
This could be

```go
return common.CancellableRecv(ctx, oc.ch)
```

Reply:
sigh... I think my brain stopped working last night. Thanks for being patient 🙇